google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rstudio/RStudioCredentialTester.java

```diff
@@ -0,0 +1,259 @@
+    }
+  }
+
+  @Override
+  public ImmutableList<TestCredential> testValidCredentials(
+      NetworkService networkService, List<TestCredential> credentials) {
+
+    return credentials.stream()
+        .filter(cred -> isRStudioAccessible(networkService, cred))
+        .collect(toImmutableList());
+  }
+
+  private boolean isRStudioAccessible(NetworkService networkService, TestCredential credential) {
+    var url = buildTargetUrl(networkService, "auth-public-key");
+    try {
+      logger.atInfo().log("Retrieving public key");
+      HttpResponse response = httpClient.send(get(url).withEmptyHeaders().build());
+      Optional<String> body = response.bodyString();
+      String exponent = body.get().split(":")[0];
+      String modulus = body.get().split(":")[1];
+
+      url = buildTargetUrl(networkService, "auth-do-sign-in");
+      logger.atInfo().log(
+          "url: %s, username: %s, password: %s",
+          url, credential.username(), credential.password().orElse(""));
+      response = sendRequestWithCredentials(url, credential, exponent, modulus);
+
+      if (response.headers().get("Set-Cookie").isPresent()
+          && response.headers().get("Set-Cookie").get().trim().startsWith("user-id")) {
+        return true;
+      } else {
+        return false;
+      }
+    } catch (IOException
+        | NoSuchProviderException
+        | NoSuchAlgorithmException
+        | BadPaddingException
+        | IllegalBlockSizeException
+        | InvalidKeyException
+        | NoSuchPaddingException
+        | InvalidKeySpecException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", url);
+      return false;
+    }
+  }
+
+  // This function is the same used from the javascript code in order to base64 encode the hexadecimal ciphertext. 
+  private String hexToBase64(String hex) {
+    StringBuilder ret = new StringBuilder();
+
+    for (int i = 0; i + 3 <= hex.length(); i += 3) {
+      int c = Integer.parseInt(hex.substring(i, i + 3), 16);
+      ret.append(B64MAP.charAt(c >> 6)).append(B64MAP.charAt(c & 63));
+    }
+
+    int remaining = hex.length() % 3;
+
+    if (remaining == 1) {
+      int c = Integer.parseInt(hex.substring(hex.length() - 1), 16);
+      ret.append(B64MAP.charAt(c << 2)).append(B64MAP);
+    } else if (remaining == 2) {
+      int c = Integer.parseInt(hex.substring(hex.length() - 2), 16);
+      ret.append(B64MAP.charAt(c >> 2)).append(B64MAP.charAt((c & 3) << 4)).append(B64PAD);
+    }
+    ret.append(B64PAD);
+    return ret.toString();
+  }
+
+  private HttpResponse sendRequestWithCredentials(
+      String url, TestCredential credential, String exponent, String modulus)
+      throws NoSuchAlgorithmException,
+          BadPaddingException,
+          IllegalBlockSizeException,
+          InvalidKeyException,
+          NoSuchPaddingException,
+          InvalidKeySpecException,
+          IOException,
+          NoSuchProviderException {
+    // encrypting with RSA PCKS#1 version 2
```

Comment:
Start comments with a capital letter and the same tense as the others ("Encrypt data with...")